.pre-commit-config.yaml

```diff
@@ -0,0 +1,28 @@
+default_language_version:
+  python: python3.11 # set for project python version
+repos:
+  - repo: local
+    hooks:
+      - id: black-apply
+        name: black-apply
+        entry: pipenv run black
+        language: system
+        pass_filenames: true
+        types: ["python"]
+      - id: mypy
+        name: mypy
+        entry: pipenv run mypy
+        language: system
+        pass_filenames: true
+        types: ["python"]
```

Comment:
We've discovered that we'll need to add the following to `mypy` hook configs:
```
exclude: "tests/"
```

More explanation here: https://mitlibraries.atlassian.net/wiki/spaces/~712020891b98ee715c482a8101f080ef16e887/pages/3434905601/Python+Projects+Pre-Commit#mypy